docs/dev/connectors/pravega.zh.md

```diff
@@ -0,0 +1,418 @@
+stream.addSink(pravegaSink);
+{% endhighlight %}
+</div>
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
+
+// Define the Pravega configuration
+val config = PravegaConfig.fromParams(params)
+
+// Define the event serializer
+val serializer = ...
+
+// Define the event router for selecting the Routing Key (lambda expression can be used for simplification)
+val router = ...
+
+// Define the sink function
+val pravegaSink = FlinkPravegaWriter.builder[MyClass]()
+    .forStream(...)
+    .withPravegaConfig(config)
+    .withSerializationSchema(serializer)
+    .withEventRouter(router)
+    .withWriterMode(EXACTLY_ONCE)
+    .build()
+
+val stream: DataStream[MyClass] = ...
+stream.addSink(pravegaSink)
+{% endhighlight %}
+</div>
+</div>
+
+#### Writer Modes
+Writer modes relate to guarantees about the persistence of events emitted by the sink to a Pravega Stream.
+The writer supports three writer modes:
+
+1. **Best-effort** - Any write failures will be ignored hence there could be data loss.
+2. **At-least-once** - All events are persisted in Pravega. Duplicate events
+are possible, due to retries or in case of failure and subsequent recovery.
+3. **Exactly-once** - All events are persisted in Pravega using a transactional approach integrated with the Flink checkpointing feature.
+For more details, please refer to [end-to-end exactly once blog](https://flink.apache.org/features/2018/03/01/end-to-end-exactly-once-apache-flink.html)
+
+By default, the _At-least-once_ option is enabled and `.withWriterMode(...)` option can be used to override the value.
+
+### Batch API
+Similarly, a `FlinkPravegaOutputFormat` class is offered for batch processing.
+It can be supplied to make a Pravega Stream as a sink of a `DataSet`.
+See the below for an example:
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// Define the output format based on a Pravega stream
+FlinkPravegaOutputFormat<EventType> outputFormat = FlinkPravegaOutputFormat.<EventType>builder()
+    .forStream(...)
+    .withPravegaConfig(config)
+    .withSerializationSchema(serializer)
+    .withEventRouter(router)
+    .build();
+
+DataSet<EventType> dataSet = ...
+dataSet.output(outputFormat);
+{% endhighlight %}
+</div>
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+// Define the output format based on a Pravega stream
+val outputFormat = FlinkPravegaOutputFormat.builder[EventType]()
+    .forStream(...)
+    .withPravegaConfig(config)
+    .withSerializationSchema(serializer)
+    .withEventRouter(router)
+    .build()
+
+val dataSet: DataSet[EventType] = ...
+dataSet.output(outputFormat)
+{% endhighlight %}
+</div>
+</div>
+
+Details on more functionality such as event time ordering, watermark propagation and metrics can be found in the
+[Document](http://pravega.io/docs/latest/connectors/connectors/index.html)
```

Comment:
broken URL, poorly named link and missing punctuation

Reply:
Fixed